src/core/reference/CMakeLists.txt

```diff
@@ -21,6 +21,17 @@ add_library(${TARGET_NAME} STATIC ${LIBRARY_SRC} ${PUBLIC_HEADERS})
 add_library(openvino::reference ALIAS ${TARGET_NAME})
 set_target_properties(${TARGET_NAME} PROPERTIES EXPORT_NAME reference)
 
+if(ENABLE_AVX2)
```

Comment:
@ilya-lavrenov 
Is add avx2 compilation flag OK, or should be handled otherwise?

Reply:
I think you need to use runtime dispatching which CPU plugin has https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/CMakeLists.txt#L282-L288

Otherwise, with current solution, you either:
- should compile w/o AVX2
- or you will have illegal instruction on non-AVX platforms like here https://github.com/openvinotoolkit/openvino/issues/26715

Reply:
Ok, will check what can be re-used.

The runtime check is done [here](https://github.com/openvinotoolkit/openvino/pull/26870/files#diff-3513a7e0516acfd1a514e89658c4cb603d2ebfc968fb844e412f5a2cac8d9325R93), already.

Compiled code from file "convert_x86_intrinsics.cpp" with avx options will be executed only if CPU has avx2 flag set

Reply:
I've double check current solution and runtime check of CPU info. And it prevents execute avx2 code if CPU will not support it. There should be errors like illegal instruction.